File: generate-launcher-icons.py
#!/usr/bin/env python3
"""
Generate Android launcher icons from IraChat logo
"""

import os
import sys
from PIL import Image, ImageDraw

def create_launcher_icons():
    """Generate all required launcher icon sizes from the IraChat logo"""
    
    # Source logo path
    logo_path = "assets/images/LOGO.png"
    
    if not os.path.exists(logo_path):
        print(f"Error: Logo file not found at {logo_path}")
        return False
    
    try:
        # Load the source logo
        logo = Image.open(logo_path)
        print(f"Loaded logo: {logo.size}")
        
        # Convert to RGBA if not already
        if logo.mode != 'RGBA':
            logo = logo.convert('RGBA')
        
        # Define all required icon sizes for Android
        icon_sizes = {
            'mipmap-mdpi': 48,
            'mipmap-hdpi': 72,
            'mipmap-xhdpi': 96,
            'mipmap-xxhdpi': 144,
            'mipmap-xxxhdpi': 192
        }
        
        # Create base directories
        base_res_path = "android/app/src/main/res"
        
        for density, size in icon_sizes.items():
            density_path = os.path.join(base_res_path, density)
            os.makedirs(density_path, exist_ok=True)
            
            # Resize logo for this density
            resized_logo = logo.resize((size, size), Image.Resampling.LANCZOS)
            
            # Create regular launcher icon
            ic_launcher_path = os.path.join(density_path, "ic_launcher.png")
            resized_logo.save(ic_launcher_path, "PNG", optimize=True)
            print(f"Created: {ic_launcher_path}")
            
            # Create foreground icon (same as regular for now)
            ic_launcher_foreground_path = os.path.join(density_path, "ic_launcher_foreground.png")
            resized_logo.save(ic_launcher_foreground_path, "PNG", optimize=True)
            print(f"Created: {ic_launcher_foreground_path}")
            
            # Create round launcher icon
            ic_launcher_round_path = os.path.join(density_path, "ic_launcher_round.png")
            
            # Create circular mask for round icon
            round_logo = Image.new('RGBA', (size, size), (0, 0, 0, 0))
            mask = Image.new('L', (size, size), 0)
            draw = ImageDraw.Draw(mask)
            draw.ellipse((0, 0, size, size), fill=255)
            
            # Apply circular mask
            round_logo.paste(resized_logo, (0, 0))
            round_logo.putalpha(mask)
            
            round_logo.save(ic_launcher_round_path, "PNG", optimize=True)
            print(f"Created: {ic_launcher_round_path}")
        
        print("\n✅ All launcher icons generated successfully!")
        return True
        
    except Exception as e:
        print(f"Error generating launcher icons: {e}")
        return False

def update_launcher_xml():
    """Update the adaptive icon XML files"""
    
    # Update ic_launcher.xml
    ic_launcher_xml = """<?xml version="1.0" encoding="utf-8"?>
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@color/ic_launcher_background"/>
    <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
</adaptive-icon>"""
    
    # Update ic_launcher_round.xml
    ic_launcher_round_xml = """<?xml version="1.0" encoding="utf-8"?>
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@color/ic_launcher_background"/>
    <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
</adaptive-icon>"""
    
    # Write XML files
    anydpi_path = "android/app/src/main/res/mipmap-anydpi-v26"
    os.makedirs(anydpi_path, exist_ok=True)
    
    with open(os.path.join(anydpi_path, "ic_launcher.xml"), 'w') as f:
        f.write(ic_launcher_xml)
    
    with open(os.path.join(anydpi_path, "ic_launcher_round.xml"), 'w') as f:
        f.write(ic_launcher_round_xml)
    
    print("✅ Updated adaptive icon XML files")

def update_colors():
    """Update colors.xml with IraChat brand color"""
    
    colors_xml = """<?xml version="1.0" encoding="utf-8"?>
<resources>
    <color name="ic_launcher_background">#87CEEB</color>
    <color name="splashscreen_background">#667eea</color>
</resources>"""
    
    colors_path = "android/app/src/main/res/values/colors.xml"
    with open(colors_path, 'w') as f:
        f.write(colors_xml)
    
    print("✅ Updated colors.xml with IraChat brand colors")

if __name__ == "__main__":
    print("🚀 Generating IraChat launcher icons...")
    
    if create_launcher_icons():
        update_launcher_xml()
        update_colors()
        print("\n🎉 All launcher icons updated successfully!")
        print("📱 The new IraChat logo will appear after rebuilding the app.")
    else:
        print("❌ Failed to generate launcher icons")
        sys.exit(1)

File: generate-launcher-icons.js
#!/usr/bin/env node
/**
 * Generate Android launcher icons from IraChat logo
 */

const fs = require('fs');
const path = require('path');

// Simple manual approach - copy the logo to all launcher locations
function copyLogoToLauncherLocations() {
    const logoPath = 'assets/images/LOGO.png';
    
    if (!fs.existsSync(logoPath)) {
        console.error(`❌ Logo file not found at ${logoPath}`);
        return false;
    }
    
    console.log('🚀 Copying IraChat logo to all launcher locations...');
    
    // Define all launcher icon locations
    const launcherLocations = [
        'android/app/src/main/res/mipmap-mdpi/ic_launcher.png',
        'android/app/src/main/res/mipmap-mdpi/ic_launcher_foreground.png',
        'android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png',
        'android/app/src/main/res/mipmap-hdpi/ic_launcher.png',
        'android/app/src/main/res/mipmap-hdpi/ic_launcher_foreground.png',
        'android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png',
        'android/app/src/main/res/mipmap-xhdpi/ic_launcher.png',
        'android/app/src/main/res/mipmap-xhdpi/ic_launcher_foreground.png',
        'android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png',
        'android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png',
        'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_foreground.png',
        'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png',
        'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png',
        'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_foreground.png',
        'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png'
    ];
    
    try {
        // Read the logo file
        const logoBuffer = fs.readFileSync(logoPath);
        
        // Copy to all launcher locations
        launcherLocations.forEach(location => {
            // Create directory if it doesn't exist
            const dir = path.dirname(location);
            if (!fs.existsSync(dir)) {
                fs.mkdirSync(dir, { recursive: true });
            }
            
            // Copy the logo
            fs.writeFileSync(location, logoBuffer);
            console.log(`✅ Updated: ${location}`);
        });
        
        console.log('\n🎉 All launcher icons updated with IraChat logo!');
        return true;
        
    } catch (error) {
        console.error(`❌ Error copying launcher icons: ${error.message}`);
        return false;
    }
}

function updateLauncherXML() {
    console.log('📝 Updating adaptive icon XML files...');
    
    const icLauncherXml = `<?xml version="1.0" encoding="utf-8"?>
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@color/ic_launcher_background"/>
    <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
</adaptive-icon>`;
    
    const icLauncherRoundXml = `<?xml version="1.0" encoding="utf-8"?>
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@color/ic_launcher_background"/>
    <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
</adaptive-icon>`;
    
    const anydpiPath = 'android/app/src/main/res/mipmap-anydpi-v26';
    
    // Create directory if it doesn't exist
    if (!fs.existsSync(anydpiPath)) {
        fs.mkdirSync(anydpiPath, { recursive: true });
    }
    
    // Write XML files
    fs.writeFileSync(path.join(anydpiPath, 'ic_launcher.xml'), icLauncherXml);
    fs.writeFileSync(path.join(anydpiPath, 'ic_launcher_round.xml'), icLauncherRoundXml);
    
    console.log('✅ Updated adaptive icon XML files');
}

function updateColors() {
    console.log('🎨 Updating colors.xml with IraChat brand colors...');
    
    const colorsXml = `<?xml version="1.0" encoding="utf-8"?>
<resources>
    <color name="ic_launcher_background">#87CEEB</color>
    <color name="splashscreen_background">#667eea</color>
</resources>`;
    
    const colorsPath = 'android/app/src/main/res/values/colors.xml';
    
    // Create directory if it doesn't exist
    const dir = path.dirname(colorsPath);
    if (!fs.existsSync(dir)) {
        fs.mkdirSync(dir, { recursive: true });
    }
    
    fs.writeFileSync(colorsPath, colorsXml);
    console.log('✅ Updated colors.xml with IraChat brand colors');
}

function removeOldWebpFiles() {
    console.log('🧹 Removing old .webp launcher files...');
    
    const webpFiles = [
        'android/app/src/main/res/mipmap-mdpi/ic_launcher.webp',
        'android/app/src/main/res/mipmap-mdpi/ic_launcher_foreground.webp',
        'android/app/src/main/res/mipmap-mdpi/ic_launcher_round.webp',
        'android/app/src/main/res/mipmap-hdpi/ic_launcher.webp',
        'android/app/src/main/res/mipmap-hdpi/ic_launcher_foreground.webp',
        'android/app/src/main/res/mipmap-hdpi/ic_launcher_round.webp',
        'android/app/src/main/res/mipmap-xhdpi/ic_launcher.webp',
        'android/app/src/main/res/mipmap-xhdpi/ic_launcher_foreground.webp',
        'android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.webp',
        'android/app/src/main/res/mipmap-xxhdpi/ic_launcher.webp',
        'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_foreground.webp',
        'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.webp',
        'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.webp',
        'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_foreground.webp',
        'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.webp'
    ];
    
    webpFiles.forEach(file => {
        if (fs.existsSync(file)) {
            fs.unlinkSync(file);
            console.log(`🗑️ Removed: ${file}`);
        }
    });
}

// Main execution
console.log('🚀 Updating IraChat launcher icons...\n');

if (copyLogoToLauncherLocations()) {
    removeOldWebpFiles();
    updateLauncherXML();
    updateColors();
    
    console.log('\n🎉 All launcher icons updated successfully!');
    console.log('📱 Rebuild the app to see the new IraChat logo in the launcher.');
    console.log('💡 Run: npx expo run:android --device');
} else {
    console.log('❌ Failed to update launcher icons');
    process.exit(1);
}
